src/hotspot/share/logging/logAsyncWriter.cpp

```diff
@@ -72,82 +110,80 @@ void AsyncLogWriter::enqueue(LogFileStreamOutput& output, LogMessageBuffer::Iter
   AsyncLogLocker locker;
 
   for (; !msg_iterator.is_at_end(); msg_iterator++) {
-    AsyncLogMessage m(&output, msg_iterator.decorations(), os::strdup(msg_iterator.message()));
-    enqueue_locked(m);
+    enqueue_locked(&output, msg_iterator.decorations(), msg_iterator.message());
   }
 }
 
 AsyncLogWriter::AsyncLogWriter()
   : _flush_sem(0), _lock(), _data_available(false),
     _initialized(false),
     _stats() {
+
+  size_t page_size = os::vm_page_size();
+  size_t size = align_up(AsyncLogBufferSize / 2, page_size);
```

Comment:
Why page size alignment? Would also interfere with my proposal of testing with a tiny buffer.

Reply:
I lift this constraint. I thought that it's was wasteful, but the functionality really doesn't depend on the page-alignment.